File: julia/space-age/space-age.jl
PLANETS = Dict(
    "Mercury" => 0.2408467,
    "Venus" => 0.61519726,
    "Earth" => 1.0,
    "Mars" => 1.8808158,
    "Jupiter" => 11.862615,
    "Saturn" => 29.447498,
    "Uranus" => 84.016846,
    "Neptune" => 164.79132,
)

EARTH_YEAR_SECONDS = 31_557_600

for (planet, orbitalperiod) in PLANETS
    eval(quote
        $(Symbol("on$planet"))(seconds) = seconds / EARTH_YEAR_SECONDS / $orbitalperiod
    end)
end

File: python/space-age/space_age.py
from functools import partial

EARTH_YEAR_SECONDS = 31_557_600


ORBITAL_PERIODS = {
    "mercury": 0.2408467,
    "venus": 0.61519726,
    "earth": 1.0,
    "mars": 1.8808158,
    "jupiter": 11.862615,
    "saturn": 29.447498,
    "uranus": 84.016846,
    "neptune": 164.79132
}


class SpaceAge:
    def __init__(self, seconds):
        for planet, orbital_period in ORBITAL_PERIODS.items():
            setattr(self, f'on_{planet}', partial(self.calculate_age, seconds, orbital_period))

    @staticmethod
    def calculate_age(seconds, orbital_period):
        return round(seconds / orbital_period / EARTH_YEAR_SECONDS, 2)